xarray/core/parallelcompat.py

```diff
@@ -0,0 +1,197 @@
+"""
+The code in this module is an experiment in going from N=1 to N=2 parallel computing frameworks in xarray.
+It could later be used as the basis for a public interface allowing any N frameworks to interoperate with xarray,
+but for now it is just a private experiment.
+"""
+from abc import ABC, abstractmethod
+from typing import Dict, Generic, Tuple, TypeVar
+
+import numpy as np
+from typing_extensions import TypeAlias
+
+from . import indexing, utils
+from .pycompat import DuckArrayModule, is_duck_dask_array
+
+T_ChunkManager = TypeVar("T_ChunkManager", bound="ChunkManager")
+T_ChunkedArray = TypeVar("T_ChunkedArray")
+T_Chunks = Tuple[Tuple[int, ...], ...]
+
+CHUNK_MANAGERS: Dict[str, T_ChunkManager] = {}
+
+
+def _get_chunk_manager(name: str) -> "ChunkManager":
+    if name in CHUNK_MANAGERS:
+        chunkmanager = CHUNK_MANAGERS[name]
+        return chunkmanager()
+    else:
+        raise ImportError(f"ChunkManager {name} has not been defined")
+
+
+class ChunkManager(ABC, Generic[T_ChunkedArray]):
+    """
+    Adapter between a particular parallel computing framework and xarray.
+
+    Attributes
+    ----------
+    array_type
+        Type of the array class this parallel computing framework provides.
+
+        Parallel frameworks need to provide an array class that supports the array API standard.
+        Used for type checking.
+    """
+
+    array_type: T_ChunkedArray
+
+    @abstractmethod
+    def chunks(self, data: T_ChunkedArray) -> T_Chunks:
+        ...
```

Comment:
@headtr1ck here I want to type hint `array_type` as the class itself, e.g. `dask.array.Array`, but I want to type the `data` arg to the method as a specific instance of that class. I kind of want something that's the inverse of `typing.Type[X]`. Is that possible? Is that a weird thing to want to do?

Reply:
Actually I think I can just use `Type[]`, but your feedback would be appreciated!

Reply:
Hey, you already found the solution :)
When you want to annotate something as a class, use `type[cls]`.

Maybe you should also use `from __future__ import annotations`, then you can use buildin `type` instead of `typing.Type` (same for dict and list etc.)

Reply:
Btw, why do you set `array_cls` in the init anyway? Will it be used in the methods that are not yet implemented? Or is it just to check if the import workd?